### PR TITLE
Pinterest Support

### PR DESCRIPTION
**Pull request by lovetostrike**
Thought this repo was no longer maintained until I saw the update. Hope these aren't huge changes. Pls let me know.

Added pinterest support.
Modified api so that users have control over order and size of social icons.
Loop to reduce code duplication.

### CODE REVIEW

**Comment by queso** [comment]
Oh wow, this is awesome. I will try to take a look at it this week.

File: client/views/social.coffee [hunk 1]
```diff
@@ -1,9 +1,6 @@
...
-    ShareIt.settings.useFB
-
-  useTwitter: () ->
-    ShareIt.settings.useTwitter
-
-  useGoogle: () ->
-    ShareIt.settings.useGoogle
+  siteTemplates: () ->
+    templates = []
+    for site in ShareIt.settings.sites
```
Comment (queso) [inline client/views/social.coffee]: This is smart, nice work!

File: client/views/social.html [hunk 2]
```diff
@@ -1,15 +1,7 @@
...
-
-  {{#if useTwitter}}
-    {{> shareit_twitter}}
-  {{/if}}
-
-  {{#if useGoogle}}
-    {{> shareit_google}}
-  {{/if}}
+  {{#each siteTemplates}}
+    {{> UI.dynamic template=this data=..}}
```
Comment (queso) [inline client/views/social.html]: :+1:

File: shareit.coffee [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
-    useGoogle: true
+    sites: [
+      'facebook',
+      'twitter',
+      'google',
+      'pinterest'
+    ]
     classes: "large btn"
     iconOnly: false
+    faSize: ''
```
Comment (queso) [inline shareit.coffee]: Does this stand for Font Awesome size?

File: tests/client.js [hunk 4]
```diff
@@ -19,13 +19,13 @@ Tinytest.add("ShareIt - {{> shareit}} template doesn't throw an exception from t
...
 Tinytest.addAsync("ShareIt - {{> shareit}} template renders valid share links", function(test, done) {
-  ShareIt.settings.useFB = true;
+  ShareIt.settings.sites.push('facebook');
   div = renderToDiv(Template.test_helper_shareit, dataContext)
   // This feels like a punt, should investigate more
   Meteor.setTimeout(function() {
     html = div.innerHTML
-    test.include(html, 'https://www.facebook.com/sharer/sharer.php?s=100&amp;p[url]=http%3A%2F%2Flocalhost%3A3000%2F&amp;p[title]=Test%20title&amp;p[summary]=undefine')
-    test.include(html, 'https://twitter.com/intent/tweet?url=http%3A%2F%2Flocalhost%3A3000%2F&amp;text=Test%20title')
+    test.include(html, 'https://www.facebook.com/sharer/sharer.php?s=100&amp;p[url]=http%3A%2F%2Flocalhost%3A10015%2Flocal&amp;p[title]=Test%20title&amp;p[summary]=undefined')
```
Comment (queso) [inline tests/client.js]: Hmm, this needs to be not hardcoded, I think, or matched slimmer without the port on it somehow...

@samhatoum - any thoughts on a better watch to do string matching here with a url inside it?

File: shareit.coffee (hunk 3 above)
Comment (lovetostrike) [inline shareit.coffee]: yes

File: tests/client.js (hunk 4 above)
Comment (lovetostrike) [inline tests/client.js]: I'm not familiar with this test setup. Travis was failing for port 3000 and returning 10015. So I just changed the port.

**Comment by rgoomar** [comment]
Nice PR!

**Pull request closed without merging**